pkg/taskservice/task_service_cron_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taskservice
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/matrixorigin/matrixone/pkg/logutil"
+	"github.com/matrixorigin/matrixone/pkg/pb/task"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+func TestScheduleCronTask(t *testing.T) {
+	runScheduleCronTaskTest(t, func(store *memTaskStorage, s *taskService, ctx context.Context) {
+		assert.NoError(t, s.CreateCronTask(ctx, newTestTaskMetadata("t1"), "*/1 * * * * *"))
+
+		s.StartScheduleCronTask()
+		defer s.StopScheduleCronTask()
+
+		time.Sleep(time.Second + fetchInterval*2)
+		tasks, err := store.Query(ctx, WithTaskParentTaskIDCond(EQ, "t1"))
+		assert.NoError(t, err)
+		assert.NotEmpty(t, tasks)
+	}, time.Millisecond, time.Millisecond)
+
+}
+
+func TestRetryScheduleCronTask(t *testing.T) {
+	runScheduleCronTaskTest(t, func(store *memTaskStorage, s *taskService, ctx context.Context) {
+		n := 0
+		store.preUpdateCron = func() error {
+			if n == 0 {
+				n++
+				return errors.New("error")
```

Comment:
:)    I would like to remove all errors.New(...)

Reply:
i will resolve this